spec/graphql/queries/users/get_users_spec.rb

```diff
@@ -0,0 +1,33 @@
+require 'rails_helper'
+
+RSpec.describe Types::QueryType do
+  describe 'display users' do
+    it 'can query all users' do
+      User.create!(name: 'Frosty', email: 'snow@brr.com', birthdate: DateTime.new(2001, 2, 3))
+      User.create!(name: 'Nick', email: 'slay@northpole.com', birthdate: DateTime.new(1950, 12, 25))
+
+      result = ApiSchema.execute(query).as_json
+      expect(result['data']['getUsers'].count).to eq(2)
+
+      users = User.all
+      expect(result.dig('data', 'getUsers')).to match_array(
+        users.map do |user|
+          { 'id' => user.id.to_s, 'name' => user.name, 'email' => user.email, 'birthdate' => user.birthdate.to_s }
+        end
+      )
+    end
+  end
+
+  def query
+    <<~GQL
+      {
+        getUsers {
+          id
+          name
+          email
+          birthdate
+          }
+      }
+    GQL
+  end
+end
```

Comment:
Very cool.  Thanks Leah!

Reply:
@leahriffell did you figure out whether these tests are hitting the test database and doing the setup and teardown? or are they hitting the production database?

Reply:
@AngelaGuardia @leahriffell This block in `rails_helper.rb` https://github.com/Turing-Eras/api/blob/main/spec/rails_helper.rb#L3-L7 should ensure that you never run tests against your production DB! 

As another note, you can also write `request` specs that simulate the request/response cycle of the FE hitting your endpoint, as opposed to `executing` your schema directly, if you want! Something like `spec/requests/graphql/user_request_spec.rb`: 
```rb
query_string = <<~GQL
   query {
     getUsers {
       id
       name
       email 
       birthdate
     }
   }
GQL

post graphql_path, params: {query: query_string}
result = JSON.parse(response.body)
expect(result).to eq({<YOUR JSON>})
```

Reply:
@Teapane @AngelaGuardia Ahhhh, thanks for showing request specs! On initial hunch, simulating a request vs. executing our schema seems like a more accurate test. Would you say that's the more suitable test for our actually testing our api?

Reply:
I think request specs make more sense when building an API that the FE is going to be consuming, since it more closely mirrors the FE making a POST request to your endpoint!

Reply:
Thanks so much for this feedback @Teapane!! Is there a case where we would want to execute as opposed to request? What I could find from the docs it seems like by executing directly we may have been skipping some query validations? @leahriffell

Reply:
I’m not as familiar with the execute pattern when testing, but could see it being valuable when asserting that types match, or conversely, when types don’t match (sad path). 🧐